src/platform/packages/private/kbn-grid-layout/grid/use_grid_layout_events/row/utils.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { GridLayoutStateManager } from '../../types';
+import { KeyboardCode, UserKeyboardEvent } from '../sensors/keyboard/types';
+
+const updateClientY = (currentY: number, stepY: number, isCloseToEdge: boolean, type = 'drag') => {
+  if (isCloseToEdge && type === 'resize') {
+    setTimeout(() => document.activeElement?.scrollIntoView({ behavior: 'smooth', block: 'end' }));
+  }
+  if (isCloseToEdge && type === 'drag') {
+    window.scrollTo({ top: window.scrollY + stepY, behavior: 'smooth' });
+    return currentY;
+  }
+  return currentY + stepY;
+};
```

Comment:
This looks to be duplicated in `src/platform/packages/private/kbn-grid-layout/grid/use_grid_layout_events/panel/utils.ts` - can this be moved to a common keyboard util?